app/services/notifications/new_comment/send.rb

```diff
@@ -17,27 +17,40 @@ def call
         return if comment.score.negative?
 
         user_ids = Set.new(comment_user_ids + subscribed_user_ids + top_level_user_ids + author_subscriber_user_ids)
+        user_ids = user_ids.delete(comment.user_id)
 
         json_data = {
           user: user_data(comment.user),
           comment: comment_data(comment)
         }
 
-        targets = []
-        user_ids.delete(comment.user_id).each do |user_id|
+        user_ids.each do |user_id|
           Notification.create(
             user_id: user_id,
             notifiable_id: comment.id,
             notifiable_type: comment.class.name,
             action: nil,
             json_data: json_data,
           )
-
-          targets << "user-notifications-#{user_id}" if User.find_by(id: user_id)&.mobile_comment_notifications
         end
 
-        # Sends the push notification to Pusher Beams channels. Batch is in place to respect Pusher 100 channel limit.
-        targets.each_slice(100) { |batch| send_push_notifications(batch) }
+        targets = User.where(id: user_ids, mobile_comment_notifications: true).ids
```

Comment:
This refactor is an enhancement to the way we were checking which PNs should be sent or not based on each of the User's `mobile_comment_notifications` setting.

**Before:** We iterated on each of the `user_ids` and queried `User.find_by(id: user_id)&.mobile_comment_notifications`. This could result in many individual queries.

**Now:** We only perform a single query to fetch the user_ids we want to target with PNs `User.where(id: user_ids, mobile_comment_notifications: true).ids`